File: learnhub-ui/src/utils/validation.ts
import * as yup from "yup";

export const password = yup
    .string()
    .min(6, "Password must have at least 6 characters")
    .matches(/[A-Z]/, "Password must have at least 1 capital letter")
    .matches(/\d/, "Password must have at least 1 digit")
    .matches(/[^a-zA-Z0-9]/, "Password must have at least 1 special character");

export const phone = yup
    .string()
    .matches(/^\d+$/, "Phone must contains only numbers")
    .min(10, "Phone must be at least 10 digits")
    .max(11, "Phone must be at most 11 digits");

File: learnhub-api/src/main/java/com/learnhub/constant/Validation.java
package com.learnhub.constant;

public interface Validation {
    String PASSWORD_REGEX = "^(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%^&*])[A-Za-z0-9!@#$%^&*]{6,}$";
    String PHONE_REGEX = "^\\d{10,11}";

    String PASSWORD_MSG = "Password must have at least 6 characters, 1 capital letter, 1 digit and 1 special character";
    String PHONE_MSG = "Phone must have 10 or 11 digits";
}
